Makefile

```diff
@@ -132,6 +132,22 @@ MIPS_VERSION := -mips2
 # we support Microsoft extensions such as anonymous structs, which the compiler does support but warns for their usage. Suppress the warnings with -woff.
 CFLAGS += -G 0 -non_shared -fullwarn -verbose -Xcpluscomm $(IINC) -nostdinc -Wab,-r4300_mul -woff 649,838,712,516
 
+# The system heap is an area of memory that covers all the VRAM that is not
+# used by compiled code/data.
+# It starts just after the end of compiled code (that is not an overlay) and
+# ends at a fixed location in VRAM.
+# That fixed location in VRAM is used by the framebuffers of the game and have
+# a fixed size.
+# This address is calculated by doing `0x80800000 - (size of framebuffers)`,
+# where 0x80800000 is the end of the Expansion Pak address range
+# This is a define in the Makefile because references in code require a raw
+# value to match (opposed to a linker symbol). This way, the define can also be
+# used in the spec file.
+SYSTEM_HEAP_END_ADDR ?= 0x80780000
```

Comment:
A few comments in general

1) the comments are nice but don't really fit in the Makefile imo. Though since we still didn't solve the discussion on where to put extended documentation I guess it can do.

2) I'm not sure it's worth abstracting so much of the memory layout instead of just having a copypasted value in two spots, if MM is anything like OoT (and I hear it's one of its best romhacks) the memory layout/usage/numbers details are sprinkled all over and you can't easily change one thing anyway unless you do know what you're doing

Reply:
Think I would agree with both points. Instead of a Makefile define, I wonder if it would just be better to add a `ifdef NON_MATCHING/`AVOID_UB` that uses the linker symbol.

Reply:
I also think it's very unusual to have an address hardcoded through the makefile.

>if MM is anything like OoT (and I hear it's one of its best romhacks)

MM is a way better "romhack" than Master Quest :)

Reply:
I agree with both points. Tbh this started like a "cool idea" I had when rearranging the spec, but it wasn't that cool when I saw it implemented :sweat_smile: 

I guess we could just simplify it to only leave a note in the spec mentioning that the user should only change that value only if they know what they are doing and telling them to also update all the other references on the repo

Reply:
Is there a reason not to put the define in a header then include that header in the spec and where it's used in code?
The spec file should be getting preprocessed so it's always an option

Reply:
I changed the spec to have a hardcoded value and moved the explanation to `buffers.h`. `Main` is still using the define
https://github.com/zeldaret/mm/pull/1488/commits/c1debe5d0ad2e389c107ccb28dbd002b55800f92